### PR TITLE
Annotate Point members usable in CUDA kernels

### PR DESCRIPTION
**Pull request by masterleinad**
Similar to #8338, this pull request introduces `CUDA` device annotations for `Point`.

### CODE REVIEW

File: tests/cuda/cuda_point.cu [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+int
+main()
+{
+  initlog();
+
+  init_cuda();
+
+  test_gpu<1, double>();
+  test_gpu<2, double>();
+  test_gpu<3, float>();
```
Comment (jppelteret) [inline tests/cuda/cuda_point.cu]:
```suggestion
  test_gpu<3, double>();
```

File: doc/news/changes/minor/20190705DanielArndt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Improved: All member funtions and free functions related to Point
```
Comment (jppelteret) [inline doc/news/changes/minor/20190705DanielArndt]:
```suggestion
Improved: All member functions and free functions related to Point
```

**Comment by masterleinad** [comment]
Thanks!

File: tests/cuda/cuda_point.cu [hunk 3]
```diff
@@ -0,0 +1,80 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
```
Comment (Rombur) [inline tests/cuda/cuda_point.cu]: Wrong year. Does the script correct the date when we add a new file?

File: tests/cuda/cuda_point.cu [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  auto p_5 = p_1 - Tensor<1, dim, Number>{};
+  auto t_1 = p_1 - p_2;
+  auto p_6 = -p_3;
+  auto p_7 = p_4 / 2.;
+  auto p_8 = p_2 * 5.;
+
+  auto s_1 = p_1 * t_1;
+  auto s_2 = p_2.square();
+  auto s_3 = p_3.distance(p_5);
+  auto s_4 = p_4.distance_square(p_1);
```
Comment (Rombur) [inline tests/cuda/cuda_point.cu]: These tests are pretty weak. As long as the code does not crash the tests will pass. You can do the checks on the device and use an array of flags to keep track that all the tests work as expected.

Comment (masterleinad) [reply tests/cuda/cuda_point.cu]: Yes. The only problems I have seen lately is compiling and running. Hence, I didn't bother testing the result to be identical with the host result so far.

File: tests/cuda/cuda_point.cu (hunk 3 above)
Comment (masterleinad) [reply tests/cuda/cuda_point.cu]: The script modifies the year the file was touched last.

**Comment by masterleinad** [comment]
I rebased and addressed the comments.

**Comment by Rombur** [comment]
CUDA code is not tested by the tester on PR. So merging as is.

**Pull request merged**